### ISSUE TITLE
Indexing wont work with = 1 or = true

### ISSUE THREAD

**Issue by jensstalder** [creof/doctrine2-spatial#178]
MySQL expects there to be no literal.

```$qb->andWhere("ST_Intersects(node.polygon, mainnode.polygon) = true");```

This works but the spacial index wont be used!

And this is not valid dql.

```$qb->andWhere("ST_Intersects(node.polygon, mainnode.polygon)");```

```... Error: Expected Literal, ...```

Do you have an idea if that can get fixed? or is there a workaround?

**Comment by yurtesen** [comment]
No, there is no workaround to this. It is related to doctrine syntax checker. See [this](https://stackoverflow.com/questions/47476822/how-do-i-use-custom-operators-in-doctrine-dql/47481297#47481297).

In addition, this issue is probably related to MySQL also because I just checked and PostgreSQL uses index. I actually switched to PostgreSQL due to MySQL's spatial support being so weak.
